File: frontend/src/app/core/services/reclamation.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { Reclamation, ReclamationStats } from '../models/reclamation.model';

@Injectable({ providedIn: 'root' })
export class ReclamationService {
  private readonly apiUrl = `${environment.apiUrl}/api/reclamations`;

  constructor(private http: HttpClient) {}

  getAll(): Observable<Reclamation[]> {
    return this.http.get<Reclamation[]>(this.apiUrl);
  }

  getById(id: number): Observable<Reclamation> {
    return this.http.get<Reclamation>(`${this.apiUrl}/${id}`);
  }

  create(reclamation: Reclamation): Observable<Reclamation> {
    return this.http.post<Reclamation>(this.apiUrl, reclamation);
  }

  update(id: number, reclamation: Reclamation): Observable<Reclamation> {
    return this.http.put<Reclamation>(`${this.apiUrl}/${id}`, reclamation);
  }

  delete(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }

  getStats(): Observable<ReclamationStats> {
    return this.http.get<ReclamationStats>(`${this.apiUrl}/stats`);
  }
}

File: backend/src/main/java/com/locavia/backend/service/AvisReclamation/ReclamationService.java
package com.locavia.backend.service.AvisReclamation;

import com.locavia.backend.dto.ReclamationDTO;

import java.util.List;
import java.util.Map;

public interface ReclamationService {

    List<ReclamationDTO> getAll();

    ReclamationDTO getById(Long id);

    ReclamationDTO create(ReclamationDTO dto);

    ReclamationDTO update(Long id, ReclamationDTO dto);

    void delete(Long id);

    Map<String, Object> getStats();
}
